dd-java-agent/agent-iast/src/main/java/com/datadog/iast/util/CookieSecurityParser.java

```diff
@@ -73,9 +73,17 @@ public static List<Cookie> parse(final HttpHeader headerName, String headerValue
           switch (state) {
             case COOKIE_NAME:
               cookieName = headerValue.substring(start, end).trim();
```

Comment:
```suggestion
              cookieName = headerValue.substring(start, next == '=' ? end - 1 : end).trim();
```
Then you can drop the extra check